pkg/goldpinger/k8s.go

```diff
@@ -41,10 +43,68 @@ func getPodNamespace() string {
 	return namespace
 }
 
+func getAllNodes() *v1.NodeList {
+	timer := GetLabeledKubernetesCallsTimer()
+	nodes, err := GoldpingerConfig.KubernetesClient.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		zap.L().Error("Error getting nodes", zap.Error(err))
+		CountError("kubernetes_api")
+	} else {
+		timer.ObserveDuration()
+	}
+	return nodes
+}
+
+// getHostIP gets the IP of the host where the pod is scheduled. If UseIPv6 is enabled then we need to check
+// the node IPs since HostIP will only list the default IP version one.
+func getHostIP(p v1.Pod) string {
+	if !GoldpingerConfig.UseIPv6 || k8snet.IsIPv6String(p.Status.HostIP) {
+		return p.Status.HostIP
+	}
+
+	nodes := getAllNodes()
```

Comment:
This might become pretty costly for a larger cluster. I wonder if we could either cache it, or take it out a level, so that we only list the nodes ones per call?

Reply:
Could you ping the pod first then collect the host ip from the response you get?

Reply:
or just always return the ipv4 host ip for now? 
In our usage of goldpinger we daemonset with host network so there is something actually listening on the host ip but I think this one is just informationaland isn't actually pinged?

Reply:
But if you really want to get this you can probably cache on host name.

Reply:
@seeker89 @paulgmiller - I changed this to get and added a basic cache for nodeName <-> hostIPv6